streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamAggWindow.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.processor.AbstractProcessor;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+
+public class KStreamAggWindow<K, V> implements ProcessorSupplier<K, V> {
+
+    @Override
+    public Processor<K, V> get() {
+        return new KStreamAggWindowProcessor();
+    }
+
+    private class KStreamAggWindowProcessor extends AbstractProcessor<K, V> {
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public void init(ProcessorContext context) {
+            super.init(context);
+        }
+
+        @Override
+        public void process(K key, V value) {
+            // create a dummy window just for wrapping the timestamp
+            long timestamp = context().timestamp();
+
+            // send the new aggregate value
+            context().forward(new Windowed<>(key, new UnlimitedWindow(timestamp)), new Change<>(value, null));
```

Comment:
Adding this Window is just for tricking the KTableProcessorSupplier which needs to provide getter view() of Windowed<K>.